src/sql/src/parsers/alter_parser.rs

```diff
@@ -248,7 +233,40 @@ impl ParserContext<'_> {
                 } else if w.keyword == Keyword::SET {
                     // consume the current token.
                     self.parser.next_token();
-                    self.parse_alter_column_set_index(column_name)
+                    let ts = self.parser.peek_token();
+                    match ts.token {
+                        Token::Word(w) if w.keyword == Keyword::DEFAULT => {
+                            self.parser.prev_token();
+                            self.parser.prev_token();
+                            self.parser.prev_token();
+                            self.parser.prev_token();
+                            // Parse `SET DEFAULT`: ALTER TABLE `table_name` MODIFY COLUMN `a` SET DEFAULT 10, MODIFY COLUMN `b` SET DEFAULT "b", ...
+                            self.parse_alter_table_set_default()
```

Comment:
Why do we step the parser backward? We already parse `modify column set/drop default` here. There is no need to parse the same thing again in `parse_alter_table_set_default` and `parse_alter_table_drop_default`.